src/main/java/dp/handler/Handler.java

```diff
@@ -46,13 +50,18 @@ public void listen(final ExportedFile message)  {
                     final ObjectMetadata metadata = new ObjectMetadata();
                     metadata.setContentLength(contentLength);
                     final String key = message.getFilterId() + ".xlsx";
-                    s3Client.putObject(new PutObjectRequest(bucket, key, new ByteArrayInputStream(xls.toByteArray()), metadata));
+                    try (final ByteArrayInputStream stream = new ByteArrayInputStream(xls.toByteArray())) {
+                        final PutObjectRequest putObjectRequest = new PutObjectRequest(bucket, key, stream, metadata);
+                        s3Client.putObject(putObjectRequest);
+                        final String downloadUri = s3Client.getUrl(bucket, key).toString();
+                        filterAPIClient.addXLSXFile(message.getFilterId().toString(), downloadUri, contentLength);
```

Comment:
I think we probably want to check the response status code from this - could output some more accurate logging if something goes wrong.